wolfcrypt/src/asn.c

```diff
@@ -23248,18 +23248,20 @@ int wc_KeyPemToDer(const unsigned char* pem, int pemSz,
     XFREE(info, NULL, DYNAMIC_TYPE_ENCRYPTEDINFO);
 #endif
 
-    if (ret < 0 || der == NULL) {
+    if (ret == 0 && der != NULL && buff == NULL) {
+        WOLFSSL_MSG("Return needed der buff length");
+        ret = der->length;
+    }
+    else if (ret < 0 || der == NULL) {
```

Comment:
Check error first.
Then, you know ret == 0 and der != NULL.

Reply:
Fixed